ddtrace/tracer/tracer_test.go

```diff
@@ -366,6 +366,32 @@ func TestSamplingDecision(t *testing.T) {
 		assert.Equal(t, "", span.context.trace.propagatingTags[keyDecisionMaker])
 		assert.Equal(t, decisionDrop, span.context.trace.samplingDecision)
 	})
+
+	t.Run("client_dropped_with_single_spans", func(t *testing.T) {
+		os.Setenv("DD_SPAN_SAMPLING_RULES", `[{"service": "test_*","name":"*_1", "sample_rate": 1.0, "max_per_second": 15.0}]`)
+		defer os.Unsetenv("DD_SPAN_SAMPLING_RULES")
+		tracer, _, _, stop := startTestTracer(t)
+		defer stop()
+		tracer.config.agent.DropP0s = true
+		tracer.config.featureFlags = make(map[string]struct{})
+		tracer.config.featureFlags["discovery"] = struct{}{}
+		tracer.config.sampler = NewRateSampler(0)
+		tracer.prioritySampling.defaultRate = 0
+		tracer.config.serviceName = "test_service"
+		parent := tracer.StartSpan("name_1").(*span)
+		child := tracer.StartSpan("name_2", ChildOf(parent.context)).(*span)
+		child.Finish()
+		parent.Finish()
+		tracer.pushTraceInfo(&traceInfo{spans: []*span{parent, child}})
+		// timing out since single span sampling runs in the worker thread
+		time.Sleep(5 * time.Millisecond)
```

Comment:
Instead of sleeping here (which could hypothetically flake) can we just stop the tracer which should trigger a flush operation? I'm not 100% sure that will work but if it does it would make sure this test doesn't flake at some point down the line

Reply:
You might also consider changing the interface of `tracer.pushTracerInfo` to accept a channel that is closed by the background goroutine when it has processed the trace.  This would facilitate similar tests.